rotkehlchen/exchanges/bitfinex.py

```diff
@@ -0,0 +1,933 @@
+            self.msg_aggregator.add_error(
+                f'Got remote error while querying {self.name} {case}: {msg}',
+            )
+            return []
+
+        raise AssertionError(f'Unexpected {self.name} response_case: {case}')
+
+    def first_connection(self) -> None:
+        self.first_connection_made = True
+
+    @protect_with_lock()
+    @cache_response_timewise()
+    def query_balances(self) -> Tuple[Optional[Dict[Asset, Balance]], str]:
+        """Return the account exchange balances on Bitfinex
+
+        The wallets endpoint returns a list where each item is a currency wallet.
+        Each currency wallet has type (i.e. exchange, margin, funding), currency,
+        balance, etc. Currencies (tickers) are in Bitfinex format and must be
+        standardized.
+
+        Endpoint documentation:
+        https://docs.bitfinex.com/reference#rest-auth-wallets
+        """
+        try:
+            currency_map_response = self._query_currency_map()
+        except IndexError:
+            msg = f'Error processing {self.name} currency map. Response is empty'
+            log.error(msg)
+            return None, msg
+
+        if currency_map_response.success is False:
+            result, msg = self._process_unsuccessful_response(
+                response=currency_map_response.response,
+                case='balances',
+            )
+            return result, msg
+
+        response = self._api_query('wallets')
+        if response.status_code != HTTPStatus.OK:
+            result, msg = self._process_unsuccessful_response(
+                response=response,
+                case='balances',
+            )
+            return result, msg
+        try:
+            response_list = rlk_jsonloads_list(response.text)
+        except JSONDecodeError as e:
+            msg = f'{self.name} returned invalid JSON response: {response.text}.'
+            log.error(msg)
+            raise RemoteError(msg) from e
+
+        # Wallet items indexes
+        currency_index = 1
+        balance_index = 2
+        asset_balance: DefaultDict[Asset, Balance] = defaultdict(Balance)
+        asset_usd_price: Dict[Asset, Price] = {}
+        for wallet in response_list:
+            if len(wallet) < API_WALLET_MIN_RESULT_LENGTH or wallet[balance_index] <= 0:
+                continue
+
+            try:
+                asset = asset_from_bitfinex(
+                    bitfinex_name=wallet[currency_index],
+                    currency_map=currency_map_response.currency_map,
+                )
+            except (UnknownAsset, UnsupportedAsset) as e:
+                asset_tag = 'unknown' if isinstance(e, UnknownAsset) else 'unsupported'
+                self.msg_aggregator.add_warning(
+                    f'Found {asset_tag} {self.name} asset {e.asset_name} due to: {str(e)}. '
+                    f'Ignoring its balance query.',
+                )
+                continue
+
+            if asset not in asset_usd_price:
```

Comment:
I can understand the idea of caching here, but I think we should make `find_usd_price()` cache recent calls instead imo. Can you open an issue? I wanted to do that for a long time.

I think I have this cache mapping too in some places and it could all be replaced by a cache in find_usd_price() itself. We even have modifiers which would do it automatically if I recall.